cmd/thanos/receive.go

```diff
@@ -407,7 +407,7 @@ func runReceive(
 	if upload {
 		// The background shipper continuously scans the data directory and uploads
 		// new blocks to Google Cloud Storage or an S3-compatible storage service.
-		bkt, err := client.NewBucket(logger, confContentYaml, reg, component.Sidecar.String())
+		bkt, err := client.NewBucket(logger, confContentYaml, reg, component.Receive.String())
```

Comment:
We have the “comp” variable that we can use here no?

Reply:
this func expects a `component.SourceStoreAPI`, not just a `component.Component`. I could type assert here or change the function signature of `runReceive`